js/base-tool.js

```diff
@@ -75,20 +117,60 @@ function baseTool () {
 					}
 
 					function getItemFromId (itId, path = "") {
+						// Get handout object, undefined if item is not a handout
 						const handout = d20.Campaign.handouts.get(itId);
 						if (handout && (handout.get("name") === CONFIG_HANDOUT || handout.get("name") === ART_HANDOUT)) return null; // skip 5etools handouts
+
+						// Get character object, undefined if item is not a character
 						const character = d20.Campaign.characters.get(itId);
+
+						// Return based on which object isn't empty
 						if (handout) return {type: "handouts", id: itId, name: handout.get("name"), path: path, archived: handout.attributes.archived};
 						if (character) return {type: "characters", id: itId, name: character.get("name"), path: path, archived: character.attributes.archived};
+
+						// If both are empty, check if item is a folder and return a folder type
+						if (d20plus.journal.checkDirExistsByPath(path.split(" / "))) return {type: "folder", id: itId, name: "", path:path, archived: false, folder: true}
 					}
 
 					function getJournalItems () {
-						if ($cbDeep.prop("checked")) return getRootJournalItems().filter(it => it);
-						else return getAllJournalItems().filter(it => it);
+						// For the root only option
+						if ($cbShallow.prop("checked")) return getRootJournalItems().filter(it => it);
```

Comment:
`filter(it => it)` can be `filter(Boolean)`, for a probably-nonexistent performance improvement 😎